From 374d13c019a0827cb5450df7f676bb5ad521d45f Mon Sep 17 00:00:00 2001
From: Oliver Brown <oliver.brown@freescale.com>
Date: Tue, 20 Nov 2012 15:29:14 -0600
Subject: [PATCH 03/26] ENGR00234203 - Fix EGL initialization failure in
 Quadrant

Commit on behalf of Chingling Wang:
The reason for failing when eglInitialize() the second time
it was called is that the display is not reset back to
ELG_NO_DISPLAY in terminate().  It will not get egl
display(eglGetDisplay()) anymore, but the egl display is already
released in terminate(). So, invalid egl display will make
eglInitialize() fail.

Signed-off-by: Oliver Brown <oliver.brown@freescale.com>
---
 opengl/libs/Android.mk          | 5 +++++
 opengl/libs/EGL/egl_display.cpp | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/opengl/libs/Android.mk b/opengl/libs/Android.mk
index 528b983..b4f9058 100644
--- a/opengl/libs/Android.mk
+++ b/opengl/libs/Android.mk
@@ -51,6 +51,11 @@ endif
 ifeq ($(TARGET_BOARD_PLATFORM), omap4)
   LOCAL_CFLAGS += -DWORKAROUND_BUG_10194508=1
 endif
+
+ifeq ($(BOARD_SOC_CLASS), IMX5X)
+  LOCAL_CFLAGS += -DIMX5X
+endif
+
 ifneq ($(MAX_EGL_CACHE_ENTRY_SIZE),)
   LOCAL_CFLAGS += -DMAX_EGL_CACHE_ENTRY_SIZE=$(MAX_EGL_CACHE_ENTRY_SIZE)
 endif
diff --git a/opengl/libs/EGL/egl_display.cpp b/opengl/libs/EGL/egl_display.cpp
index 26240f1..f1bed2b 100644
--- a/opengl/libs/EGL/egl_display.cpp
+++ b/opengl/libs/EGL/egl_display.cpp
@@ -257,6 +257,11 @@ EGLBoolean egl_display_t::terminate() {
         res = EGL_TRUE;
     }
 
+#if defined(IMX5X)
+    egl_cache_t::get()->terminate();
+    disp.dpy = EGL_NO_DISPLAY;
+#endif
+	
     mHibernation.setDisplayValid(false);
 
     // Reset the extension string since it will be regenerated if we get
-- 
1.8.0

